Default export doesn't work with project references

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:** 3.0, 3.1.0-dev.20180824

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:**

Project References, default export

**Code**

Repository: https://github.com/ikokostya/ts-proj-ref-bug

Project structure:

```
foo/
   test.ts
   tsconfig.json
bar/
   test.ts
   tsconfig.json
```

```ts
// foo/test.ts

interface Foo {
    x: number;
}

export default Foo;
```

```json5
// foo/tsconfig.json 
{
    "compilerOptions": {
        "composite": true,
        "outDir": "../out/foo"
    },
    "include": [
        "*.ts"
    ]
}
```

```ts
// bar/test.ts

import Foo from '../foo/test';
```

```json5
// bar/tsconfig.json 
{
    "compilerOptions": {
        "composite": true,
        "outDir": "../out/bar"
    },
    "include": [
        "*.ts"
    ],
    "references": [
        {"path": "../foo"}
    ]
}
```

**Expected behavior:**

No errors.

**Actual behavior:**

Compile error:

```
out/foo/test.d.ts:1:16 - error TS2304: Cannot find name 'Foo'.

1 export default Foo;
                 ~~~
```

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior -->

**Related Issues:** https://github.com/Microsoft/TypeScript/issues/25600

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -26347,7 +26347,7 @@ namespace ts {
 
         function checkExportSpecifier(node: ExportSpecifier) {
             checkAliasSymbol(node);
-            if (compilerOptions.declaration) {
+            if (getEmitDeclarations(compilerOptions)) {
                 collectLinkedAliases(node.propertyName || node.name, /*setVisibility*/ true);
             }
             if (!node.parent.parent.moduleSpecifier) {
@@ -26388,7 +26388,7 @@ namespace ts {
             if (node.expression.kind === SyntaxKind.Identifier) {
                 markExportAsReferenced(node);
 
-                if (compilerOptions.declaration) {
+                if (getEmitDeclarations(compilerOptions)) {
                     collectLinkedAliases(node.expression as Identifier, /*setVisibility*/ true);
                 }
             }

--- a/tests/baselines/reference/declarationEmitWithComposite.js
+++ b/tests/baselines/reference/declarationEmitWithComposite.js
@@ -11,4 +11,7 @@ exports.__esModule = true;
 
 
 //// [/foo/out/test.d.ts]
+interface Foo {
+    x: number;
+}
 export default Foo;

--- a/tests/baselines/reference/declarationEmitWithComposite.js
+++ b/tests/baselines/reference/declarationEmitWithComposite.js
@@ -1,0 +1,14 @@
+//// [test.ts]
+interface Foo {
+    x: number;
+}
+export default Foo;
+
+
+//// [/foo/out/test.js]
+"use strict";
+exports.__esModule = true;
+
+
+//// [/foo/out/test.d.ts]
+export default Foo;

--- a/tests/baselines/reference/declarationEmitWithComposite.symbols
+++ b/tests/baselines/reference/declarationEmitWithComposite.symbols
@@ -1,0 +1,10 @@
+=== /foo/test.ts ===
+interface Foo {
+>Foo : Symbol(Foo, Decl(test.ts, 0, 0))
+
+    x: number;
+>x : Symbol(Foo.x, Decl(test.ts, 0, 15))
+}
+export default Foo;
+>Foo : Symbol(Foo, Decl(test.ts, 0, 0))
+

--- a/tests/baselines/reference/declarationEmitWithComposite.types
+++ b/tests/baselines/reference/declarationEmitWithComposite.types
@@ -1,0 +1,8 @@
+=== /foo/test.ts ===
+interface Foo {
+    x: number;
+>x : number
+}
+export default Foo;
+>Foo : Foo
+

--- a/tests/cases/compiler/declarationEmitWithComposite.ts
+++ b/tests/cases/compiler/declarationEmitWithComposite.ts
@@ -1,0 +1,13 @@
+// @composite: true
+// @fullEmitPaths: true
+
+// @filename: /foo/tsconfig.json
+{
+    "compilerOptions": { "composite": true, "outDir": "out" }
+}
+
+// @filename: /foo/test.ts
+interface Foo {
+    x: number;
+}
+export default Foo;

